### ISSUE TITLE
calc_*() functions that can take vectors or variable names

### ISSUE THREAD

**Issue by jepusto** [meghapsimatrix/simhelpers#7]
It seems like it would be useful to allow the calc_*() functions to operate either on variable names, within a data.frame as specified in res_dat, or _directly on vectors.  Paired with dplyr 1.0, the latter would then circumvent the need for `do()` and allow for the following syntax:
```
results %>%
    group_by(method) %>%
    mutate(calc_absolute(estimates = est, true_param = mean_diff))

results %>%
    group_by(method) %>%
    mutate(calc_rejection(p_values = p_val))
```

**Comment by meghapsimatrix** [comment]
Got it. Will work on it. I should prob wait till dplyr 1.0 releases right? When you say directly on vectors, do you mean take res_dat out as an argument?

**Comment by jepusto** [comment]
I think there is probably a way to write it so that the `res_dat` argument
is _optional_. See the chapter on non-standard evaluation in Hadley's
Advanced R book for pointers. If it's optional, then there's no need to
wait for dplyr 1.0.

**Comment by jepusto** [comment]
That said, I classified this as an enhancement because I don't think it's needed for first release to CRAN.

**Comment by meghapsimatrix** [comment]
Okay I will try to do it. Thanks!

**Comment by meghapsimatrix** [comment]
You wanted something like this right? from dplyr 1.0.0
![image](https://user-images.githubusercontent.com/33493716/77694464-0b9fc480-6f78-11ea-8507-d8778a24c373.png)

I tried somethings but it def doesn't work with mutate and currently doesn't work with summarize as the calc_* functions return more than one column. I tried to output only one column. It works on grouped data with summarize but ugly. I have dplyr 0.8.5.
https://github.com/meghapsimatrix/simhelpers/blob/master/auxiliary/make_res_dat_optional.R

I read the chapter from Advanced R but not quite sure how to use that info here.
